src/lib.rs

```diff
@@ -99,12 +95,10 @@ impl PID {
         self.err_sum = self.err_sum + err;
         // Error rate of change for derivative portion
         let mut err_dt: f64 = 0.0;
-        // Ignore D value on first step (timer will be None)
-        if let Some(timer) = self.timer {
-            let dt = timer.elapsed().as_secs_f64();
+        // Ignore D value on first step (dt will be None)
+        if let Some(dt) = dt_s {
             err_dt = (err - self.err_prev) / dt;
         }
```

Comment:
`err_dt` really doesn't need to be `mut`:

```suggestion
        // Ignore D value on first step (dt will be None)
        let err_dt = if let Some(dt) = dt_s {
            (err - self.err_prev) / dt;
        } else {
            0.0
        }
```